### PR TITLE
fix: advance matching in the route create page causes the page to crash

### LINKED ISSUE #2438: bug: advance matching in the route create page causes the page to crash
### Issue description

When we create an advanced match rule in a route, the page will crash if the reverse setting is not adjusted.

Based on the occurrence conditions, it is presumed that the default value of the reverse field is not set correctly. At that point, executing toString on a `null` value will result in a JS error.

https://github.com/apache/apisix-dashboard/blob/1fa5df3618708b392bb5fc2c866419b406cc43a7/web/src/pages/Route/components/Step1/MatchingRulesView.tsx#L148-L152

### Expected behavior

correct operation

### How to Reproduce

1. go to route create page
2. create a advance matching rule that not change `reverse` field (**don't click it**)
3. click `confirm`
4. page crashed

### Screenshots

_No response_

### Environment

- apisix-dashboard version, if have: 2.11.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Si-ege**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #2438
1.fix matching in the route create page causes the page to crash

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by Si-ege** [comment]
My understanding of this issue is:
1. the method toString is not executed when the default value of the reverse field is undefined

**Comment by Baoyuantop** [comment]
Hi @Si-ege, we need E2E test for this change:
1. Go to Route List and click Create
2. Fill in the necessary information and click the `Add` button in the Advanced Routing Matching Conditions section.
3. Fill in everything in the Create Rule except `Reverse the result(!)` field
4. Click Confirm and we can see the data in the table and `Reverse the result(!)` field is false.
5. We can create this route successfully

**Comment by Si-ege** [comment]
> ok

**Comment by Baoyuantop** [comment]
Hi @Si-ege, thank you for your contribution. For the E2E test section, we would prefer to create a new test case, meaning to write the test in a new `it()` instead of changing the previous test.

File: web/cypress/integration/route/can-skip-upstream-when-select-service-id.spec.js [hunk 1]
```diff
@@ -103,12 +112,45 @@ context('Can select service_id skip upstream in route', () => {
...
+    });
+    cy.get('.ant-form-item-control-input-content > #name').type(data.parameterName);
+    cy.get(selector.operator).click();
+    cy.get(selector.selectItem).within(() => {
+      cy.contains('IN').click();
+    });
+    cy.get(selector.value).type(data.value);
+    cy.contains('Confirm').click();
+    cy.get(selector.rowcard).should('be.visible');
+    cy.get(selector.rowcard).get('tr>td').eq(2).should('have.value', '');
```
Comment (Baoyuantop) [inline web/cypress/integration/route/can-skip-upstream-when-select-service-id.spec.js:143]: Should it show up as false here?

File: web/src/pages/Route/components/Step1/MatchingRulesView.tsx [hunk 2]
```diff
@@ -148,7 +148,8 @@ const MatchingRulesView: React.FC<RouteModule.Step1PassProps> = ({
     {
       title: formatMessage({ id: 'page.route.reverse' }),
       key: 'reverse',
-      render: (text: RouteModule.MatchingRule) => text.reverse.toString(),
+      render: (text: RouteModule.MatchingRule) =>
+        text.reverse === undefined ? text.reverse : text.reverse.toString(),
```
Comment (Baoyuantop) [inline web/src/pages/Route/components/Step1/MatchingRulesView.tsx:152]: I think it is more appropriate to set an initial value for reverse in the Form.

**Comment by guoqqqi** [comment]
ping @Si-ege  any update?

**Review by Baoyuantop — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**